src/main.rs

```diff
@@ -107,28 +126,127 @@ enum ClientEventGame {
     Answer { vote: Vote },
 }
 
-lazy_static::lazy_static! {
-    static ref ROOMS: Mutex<HashMap<String, Room<GameRoom>>> = Mutex::new(HashMap::new());
-    static ref QUESTIONS: Vec<Prompt> = {
-        // TODO: c pa opti
-        let content = std::fs::read_to_string("questions.ron").expect("Error while reading question.ron file");
-        let questions: Vec<Prompt> = ron::from_str(&content).expect("Error while parsing questions");
-
-        println!("Read {} questions ({} tags)", questions.len(), questions.iter().filter(|x| match x {
-            Prompt::Tag(_, _) => true,
-            _ => false,
-        }).count());
-
-        questions
-    };
-}
-
 impl From<&ServerEvent<'_>> for Message {
     fn from(event: &ServerEvent<'_>) -> Self {
         Message::Text(serde_json::to_string(event).unwrap())
     }
 }
 
+struct Lobby {
+    funny_words: FunnyWords,
+    questions: Arc<[Prompt]>,
+    rooms: HashMap<String, Room<GameRoom>>,
+}
+
+impl RoomHandler for Lobby {
+    type Guest = ();
+
+    fn on_message(&mut self, cx: Context<Self::Guest>, msg: Message) -> ResultRelocation {
+        use ServerEvent::*;
+
+        let msg = match msg {
+            Message::Text(s) => s,
+            _ => unreachable!(),
+        };
+
+        let evt = match serde_json::from_str(&msg) {
+            Ok(msg) => msg,
+            _ => return Ok(None),
+        };
+
+        match evt {
+            ClientEventLobby::RoomProbe { code } => {
+                let code = code.as_str();
+
+                #[cfg(debug_assertions)]
+                if code == "TEST" {
+                    cx.send(&ServerEvent::RoomProbeResult {
+                        code: Some(code),
+                    })?;
+
+                    return Ok(None)
+                }
```

Comment:
Idk, I intentionally left it (with `#[cfg(debug_assertions)] but it might not be useful in the long run... It just adds a few lines and makes it easier to test the probing system.

Reply:
Yes but it means that any debug build will have a broken probing system

Reply:
Why "broken" ? Real room codes are always 8 characters long.
Anyway, I'll remove it.